src/webgl/p5.Framebuffer.js

```diff
@@ -931,6 +950,238 @@ class Framebuffer {
+      colorFormat.format,
+      colorFormat.type
+    );
+    gl.bindFramebuffer(gl.FRAMEBUFFER, prevFramebuffer);
+  }
+
+  /**
+   * Get a region of pixels, or a single pixel, from the canvas.
+   *
+   * Returns an array of [R,G,B,A] values for any pixel or grabs a section of
+   * an image. If the Framebuffer has been set up to not store alpha values, then
+   * only [R,G,B] will be returned. If no parameters are specified, the entire
+   * image is returned.
+   * Use the x and y parameters to get the value of one pixel. Get a section of
+   * the display window by specifying additional w and h parameters. When
+   * getting an image, the x and y parameters define the coordinates for the
+   * upper-left corner of the image, regardless of the current <a href="#/p5/imageMode">imageMode()</a>.
+   *
+   * @method get
+   * @param  {Number}         x x-coordinate of the pixel
+   * @param  {Number}         y y-coordinate of the pixel
+   * @param  {Number}         w width
+   * @param  {Number}         h height
+   * @return {p5.Image}       the rectangle <a href="#/p5.Image">p5.Image</a>
+   */
+  /**
+   * @method get
+   * @return {p5.Image}      the whole <a href="#/p5.Image">p5.Image</a>
+   */
+  /**
+   * @method get
+   * @param  {Number}        x
+   * @param  {Number}        y
+   * @return {Number[]}      color of pixel at x,y in array format [R, G, B, A]
+   */
+  get(x, y, w, h) {
+    const colorFormat = this._glColorFormat();
+    if (x === undefined && y === undefined) {
+      x = 0;
+      y = 0;
+      w = this.width;
+      h = this.height;
+    } else if (w === undefined && h === undefined) {
+      if (x < 0 || y < 0 || w >= this.width || h >= this.height) {
+        return [0, 0, 0, 0];
+      }
+
+      return readPixelWebGL(
+        this.gl,
+        this.framebuffer,
+        x * this.density,
+        y * this.density,
+        colorFormat.format,
+        colorFormat.type
+      );
+    }
+
+    const rawData = readPixelsWebGL(
+      undefined,
+      this.gl,
+      this.framebuffer,
+      x * this.density,
+      y * this.density,
+      w * this.density,
+      h * this.density,
+      colorFormat.format,
+      colorFormat.type
+    );
+    // Framebuffer data might be either a Uint8Array or Float32Array
+    // depending on its format, and it may or may not have an alpha channel.
+    // To turn it into an image, we have to normalize the data into a
+    // Uint8ClampedArray with alpha.
+    const fullData = new Uint8ClampedArray(
+      w * h * this.density * this.density * 4
+    );
+    const channels = colorFormat.type === this.gl.RGB ? 3 : 4;
+    for (let y = 0; y < h * this.density; y++) {
+      for (let x = 0; x < w * this.density; x++) {
+        for (let channel = 0; channel < 4; channel++) {
+          const idx = (y * w * this.density + x) * 4 + channel;
+          if (channel >= channels) {
+            fullData[idx] = 255;
+          } else {
+            const prevIdx = channels === 4
```

Comment:
technically not really a previndex but more the rgb index right? Can the variable naming make this distiction clearer to future maintainers?